Issue #563

stepanzharychevbroadcom opened the pull request.

VitGottwald commented on src/utils/workspace.ts:
```diff
@@ -0,0 +1,27 @@
+    id: string;
+}
+
+export async function checkTextFileIsOpened(path: string) {
```
Are we sure this does not interfere with the user work flow - like blinking a window or activating some other window than currently selected?

stepanzharychevbroadcom replied:
It will be, as I showed you, but unfortunately there's no other way for us to achieve the requested functionality.

stepanzharychevbroadcom commented:
> Thanks Stepan an interesting solution. Needs test coverage please
> <img alt="image" width="585" src="https://user-images.githubusercontent.com/30794003/76296292-e2abde00-62ad-11ea-8738-86dc6655d8e5.png">

@Colin-Stone Should be fixed now.

stepanzharychevbroadcom commented:
@Colin-Stone @VitGottwald @zFernand0 
What deserves special attention in this review: **workspace.ts**.
I introduced the fix which omits empty/binary tabs up to 3 times, and after stops. (this condition required not to go to the infinite loop) It's reflected in tests.

Also I added **constants**, so magic values can be used across the application, not stay hardcoded in 1 file.

stepanzharychevbroadcom commented:
New implementation is in https://github.com/zowe/vscode-extension-for-zowe/pull/817. PR should be closed.
